### PR TITLE
HashPack wallet integration

### PR DESCRIPTION
**Pull request by SarjuHansaliya**
1. HashPack Wallet Integration
2. Implement few hooks related on Hedera
3. Add HederaProvider

### CODE REVIEW

File: src/connectors/HashConnector/index.ts [hunk 1]
```diff
@@ -151,8 +153,28 @@ export class HashConnector extends AbstractConnector {
...
 
   public async getAccount(): Promise<null | string> {
+    console.log('getAccount', this.pairingData);
     if (this.pairingData) {
-      return this.pairingData?.accountIds[0];
+      try {
+        const newAccountId = this.pairingData?.accountIds[0];
+        const operatorAccount = '0.0.47977330';
+        const operatorPrivateKey =
+          '302e020100300506032b6570042204208b97c5c963a2ffa06ca9c3a5837eaebc9d21fd92c42a8b79baa2c13af306fa53';
```
Comment (HariSeldon23) [inline src/connectors/HashConnector/index.ts:162]: What does this represent? Surely this should be an environmental variable?

**Comment by HariSeldon23** [comment]
We need to add the following controls to the WallectModal:
* trackWallets - boolean (this won't do anything for now)
* supportedWallets - enum

Then based upon which Wallets are chosen, we will then only display those

**Comment by HariSeldon23** [comment]
Another control that needs to be added would be 
* topicURL
* topicData

This will then mean that when a user connects to the wallet we should send some data to the topicURL. The topicData should look like the following:
{
    walletConnectedTo: Metamask,
    vmType: EVM,
    address: "0x23829382938298392832",
    partnerID: "2342323"
}

We will use this to push data into Google PubSub and then once it's been published, we'll have some data pipelines pick it up and run some analytics

File: src/connectors/HashConnector/index.ts (hunk 1 above)
Comment (SarjuHansaliya) [reply src/connectors/HashConnector/index.ts]: this was just for testing purpose.

**Pull request merged**